### ISSUE TITLE
Unused wild card imports are not removed

### ISSUE THREAD

**Issue by scottresnik** [diffplug/spotless#240]
Wild card imports seem to be ignored if they are unused.  Is it possible to correct this?  At the very least would it be possible to add a setting to fail if wild card imports are encountered?

- Gradle 4.7
- Spotless 3.10.0
- Windows 10 Pro

**Comment by scottresnik** [comment]
[Example on github.](https://github.com/scottresnik/spotless-wildcard/blob/0a4aa926951b8bfe342f66d05a6c3695752e82c7/src/test/java/com/foo/lib/LibraryTest.java#L6)

**Comment by nedtwigg** [comment]
Consider this situation:

```java
import pkgA.*;
import pkgB.*;

public class Clazz {
    WhatPackageDidIComeFrom field;
}
```

Without slurping in the classpath, and then looking at the classfiles in those jars, we cannot know which package it came from.  Basically requires a full compile.  With #239 it might be vaguely possible to do, but it will certainly be a lot of work.

**Comment by scottresnik** [comment]
I kind of figured that would be the case.

What do you think about a setting to fail on cases of wild card imports?

**Comment by nedtwigg** [comment]
You can use a custom rule, something like:

```gradle
custom 'noWildcardImports', {
  if (it.contains('*;\n')) {
    throw new Error('No wildcard imports allowed')
  }
}
bumpThisNumberIfACustomStepChanges(1)
```

**Comment by nedtwigg** [comment]
Sounds good.  Spotless is optimized for code formatting of the form `String -> String`.  If you want to disallow wildcards, you can fudge it with Spotless, or use another tool.  I don't think it makes sense to integrate static code analysis with Spotless unless it produces the `String -> String` output required for `spotlessApply` to work.

**Comment by nedtwigg** [comment]
Released in `plugin-gradle 7.1.0` and `plugin-maven 2.45.0`.
